docs/dev/task_failure_recovery.md

```diff
@@ -1,5 +1,5 @@
 ---
-title: "Restart Strategies"
+title: "Task Failure Recovery"
```

Comment:
Agree. The batch fault tolerance page is suggesting a deprecated way to configure restart strategies. Deleting it can also help with further deprecating it.

Reply:
I opened a ticket for it: https://issues.apache.org/jira/browse/FLINK-13336 .
Can start working on it once this PR is merged.